pkg/inclusion/paths_test.go

```diff
@@ -338,97 +340,144 @@ func Test_genSubTreeRootPath(t *testing.T) {
-		//
-		// |       | blob  | blob  |       |
-		// 128 ... 144 ... 160 ... 176 ... 192
-		{64, 144, 32, []path{
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkLeft}, row: 2},
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkRight}, row: 2},
-			{instructions: []WalkInstruction{WalkRight, WalkLeft, WalkLeft}, row: 2},
-			{instructions: []WalkInstruction{WalkRight, WalkLeft, WalkRight}, row: 2},
-		}},
-		// first 33 shares in the last row of a 64 x 64 square.
-		{64, 4032, 33, []path{
-			{instructions: []WalkInstruction{WalkLeft, WalkLeft, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkLeft, WalkRight}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkRight}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkLeft, WalkLeft, WalkLeft, WalkLeft, WalkLeft}, row: 63},
-		}},
-		// last 63 shares in the last row of a 64 x 64 square.
-		{64, 4032, 63, []path{
-			{instructions: []WalkInstruction{WalkLeft, WalkLeft, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkLeft, WalkRight}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkLeft, WalkRight, WalkRight}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkLeft, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkLeft, WalkRight}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkRight, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkRight, WalkRight, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkRight, WalkRight, WalkRight, WalkLeft}, row: 63},
-			{instructions: []WalkInstruction{WalkRight, WalkRight, WalkRight, WalkRight, WalkRight, WalkLeft}, row: 63},
-		}},
+		{
+			"all paths for a basic 2x2", 2, 2, 2,
+			[]path{
+				{
+					row:          1,
+					instructions: []WalkInstruction{WalkLeft},
+				},
+				{
+					row:          1,
+					instructions: []WalkInstruction{WalkRight},
+				},
+			},
+			[]int{0, 1},
+		},
+		{
+			"all paths for a basic 4x4", 4, 2, 2,
+			[]path{
+				{
+					row:          0,
+					instructions: []WalkInstruction{WalkRight, WalkLeft},
+				},
+				{
+					row:          0,
+					instructions: []WalkInstruction{WalkRight, WalkRight},
+				},
+			},
+			[]int{0, 1},
+		},
+		{
+			"all paths for a basic 4x4 span more than 1 row", 4, 3, 2,
+			[]path{
+				{
+					row:          0,
+					instructions: []WalkInstruction{WalkRight, WalkRight},
+				},
+				{
+					row:          1,
+					instructions: []WalkInstruction{WalkLeft, WalkLeft},
+				},
+			},
+			[]int{0, 1},
+		},
+		{
+			"single share in the middle of a 128x128", 128, 8252, 1,
+			[]path{
+				{
+					row:          64,
+					instructions: []WalkInstruction{WalkLeft, WalkRight, WalkRight, WalkRight, WalkRight, WalkLeft, WalkLeft},
```

Comment:
[for reviewers]

We're looking for index 60 of 128 on row 64

Initially the Merkle root spans leaf nodes: 0-128
1. Left: 0-64
2. Right: 32-64
3. Right: 48-64
4. Right: 56-64
5. Right: 60-64
6. Left: 60-62
7. Left: 60-61